internal/provider/resource/integration_aws_parameter_store.go

```diff
@@ -286,6 +311,15 @@ func (r *IntegrationAWSParameterStoreResource) Read(ctx context.Context, req res
 	if resp.Diagnostics.HasError() {
 		return
 	}
+	_, err := pkg.ValidateAwsInputCredentials(state.AccessKeyID, state.SecretAccessKey, state.AssumeRoleArn)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error validating AWS credentials",
+			err.Error(),
+		)
+		return
+	}
```

Comment:
do we need this part?

not sure why we're validating configuration from inside the `Read` method